threads/threads.c

```diff
@@ -245,6 +247,9 @@ static void os_static_mutex_unlock(os_static_mutex_t *s)
      LONG old = InterlockedExchange(s, 0);
      A(old == 1);
 }
+
+#define NOT_SIGNALED (WaitForSingleObject(q->ready, 0) == WAIT_TIMEOUT)
```

Comment:
Perhaps you should try waiting for a thread here, not the semaphore, something like this: `#define NOT_SIGNALED (WaitForSingleObject(m_Threads[0], 1) == WAIT_OBJECT_0)`

Reply:
@dzenanz I don't know what's best but between testing the thread or the mutex. Can you explain your suggestion? Also, why 1 and not 0 for the timeout?